src/main/java/org/apache/xml/security/utils/resolver/ResourceResolverContext.java

```diff
@@ -51,4 +57,12 @@ public ResourceResolverContext(Attr attr, String baseUri, boolean secureValidati
         return properties;
     }
 
+    public boolean isSafeURIToResolve() {
+        if (allowUnsafeResourceResolving) {
+            return true;
+        }
+        boolean forbiddenURI = (uriToResolve != null && (uriToResolve.startsWith("file:") || uriToResolve.startsWith("http:")))
```

Comment:
Unless this is done somewhere else, I think you should lowercase `uriToResolve` (probably in the ctor) so that it also matches mixed/upper case schemes, ex "HTTP:".

Reply:
I think we don't need to do this, as the resource resolvers only handle lower case file/http URLs